bigstream/features.py

```diff
@@ -40,18 +44,60 @@ def blob_detection(
         detected blobs. The last column is the image intensity at the
         detected coordinate location.
     """
-
     processed_image = np.copy(image)
-    if winsorize_limits is not None:
-        processed_image = winsorize(processed_image, limits=winsorize_limits)
+    start_time = time.time()
+
+    # ensure iterable radii
+    if not isinstance(min_blob_radius, (tuple, list, np.ndarray)):
+        min_blob_radius = (min_blob_radius,)*image.ndim
+    if not isinstance(max_blob_radius, (tuple, list, np.ndarray)):
+        max_blob_radius = (max_blob_radius,)*image.ndim
+
+    blob_detect_method = None
+    if blob_method == 'dog':
+        # difference of gaussian
+        blob_detect_method = blob_dog
+    else:
+        # laplacian of gaussian
+        blob_detect_method = blob_log
+        kwargs['num_sigma'] = num_sigma
+
+    # set given arguments
+    kwargs['min_sigma'] = np.array(min_blob_radius) / np.sqrt(image.ndim)
+    kwargs['max_sigma'] = np.array(max_blob_radius) / np.sqrt(image.ndim)
+
+    # set additional defaults
+    if 'threshold' not in kwargs or kwargs['threshold'] is None:
+        kwargs['threshold'] = None
+        kwargs['threshold_rel'] = 0.1
+
+    print(f'{time.ctime(time.time())}',
+          f'Start spot detection ({min_blob_radius}, {max_blob_radius})',
+          kwargs,
+          flush=True)
+    if winsorize_limits:
+        processed_image = winsorize(processed_image, limits=winsorize_limits,
+                                    inplace=True)
+        done_winsorize_time = time.time()
+        print(f'{time.ctime(time.time())}',
+              f'Winsorization completed in {done_winsorize_time-start_time}s',
+              flush=True)
     if background_subtract:
+        done_tophat_time = time.time()
```

Comment:
I think you want this `done_tophat_time` to be after the call to the tophat function right?

Reply:
correct